### PR TITLE
Add a kitten notifier

### PR DESCRIPTION
**Pull request by 4444N**

### CODE REVIEW

File: runelite-client/src/main/java/net/runelite/client/plugins/kittennotifier/KittenNotifierConfig.java [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+            description = "Only notify when kitten absolutely needs food or attention."
+    )
+    default boolean absolutelyNeeded() { return false; }
+    @ConfigItem(
+            keyName = "catOwned",
+            name = "",
+            description = "",
+            hidden = true
+    )
+    default boolean catOwned() { return false; }
```
Comment (Spedwards) [inline runelite-client/src/main/java/net/runelite/client/plugins/kittennotifier/KittenNotifierConfig.java]: Braces on new line

```suggestion
	default boolean catOwned()
	{
		return false;
	}
```

File: runelite-client/src/main/java/net/runelite/client/plugins/kittennotifier/KittenNotifierPlugin.java [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+    @Inject
+    private Client client;
+    @Provides
+    KittenNotifierConfig getConfig(ConfigManager configManager)
+    {
+        return configManager.getConfig(KittenNotifierConfig.class);
+    }
+    @Subscribe
+    public void onChatMessage(ChatMessage event) {
+        if (event.getType() == ChatMessageType.SERVER && !config.catOwned()) {
```
Comment (Spedwards) [inline runelite-client/src/main/java/net/runelite/client/plugins/kittennotifier/KittenNotifierPlugin.java]: Same as before, braces on new lines.

File: runelite-api/src/main/java/net/runelite/api/widgets/WidgetID.java [hunk 3]
```diff
@@ -148,7 +148,6 @@
 		static final int CONTINUE = 3;
 		static final int TEXT = 4;
 	}
-
```
Comment (Spedwards) [inline runelite-api/src/main/java/net/runelite/api/widgets/WidgetID.java]: Do not remove this line.

File: runelite-client/src/main/java/net/runelite/client/plugins/kittennotifier/KittenNotifierConfig.java [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+@ConfigGroup("kittennotifier")
+public interface KittenNotifierConfig extends Config{
+	@ConfigItem
+	(
+		keyName = "absolutelyNeeded",
+		name = "Notify only on Absolute Need",
+		description = "Only notify when kitten absolutely needs food or attention."
+	)
+	default boolean absolutelyNeeded() { return false; }
+	@ConfigItem
```
Comment (Spedwards) [inline runelite-client/src/main/java/net/runelite/client/plugins/kittennotifier/KittenNotifierConfig.java]: Keep a blank line between each item. See other config files for reference.

File: runelite-client/src/main/java/net/runelite/client/plugins/kittennotifier/KittenNotifierPlugin.java [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+
+@PluginDescriptor(
+	name = "Kitten Notifier",
+	description = "Sends a notification when your kitten needs food, attention, or is grown.",
+	tags = {"kitten, notifications"}
+)
+public class KittenNotifierPlugin extends Plugin{
+	@Inject
+	private Notifier notifier;
+	@Inject
```
Comment (Spedwards) [inline runelite-client/src/main/java/net/runelite/client/plugins/kittennotifier/KittenNotifierPlugin.java]: Keep a blank line between injected fields. See other classes for reference.

File: runelite-client/src/main/java/net/runelite/client/plugins/kittennotifier/KittenNotifierPlugin.java [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+	private KittenNotifierConfig config;
+	@Inject
+	private Client client;
+	@Provides
+	KittenNotifierConfig getConfig(ConfigManager configManager)
+	{
+		return configManager.getConfig(KittenNotifierConfig.class);
+	}
+	@Subscribe
```
Comment (Spedwards) [inline runelite-client/src/main/java/net/runelite/client/plugins/kittennotifier/KittenNotifierPlugin.java]: You get the point, blank lines.

File: runelite-client/src/main/java/net/runelite/client/plugins/kittennotifier/KittenNotifierConfig.java [hunk 7]
```diff
@@ -0,0 +1,38 @@
+package net.runelite.client.plugins.kittennotifier;
+import net.runelite.client.config.Config;
+import net.runelite.client.config.ConfigGroup;
+import net.runelite.client.config.ConfigItem;
+
+@ConfigGroup("kittennotifier")
+public interface KittenNotifierConfig extends Config{
```
Comment (Spedwards) [inline runelite-client/src/main/java/net/runelite/client/plugins/kittennotifier/KittenNotifierConfig.java]:
```suggestion
public interface KittenNotifierConfig extends Config
{
```

File: runelite-client/src/main/java/net/runelite/client/plugins/kittennotifier/KittenNotifierPlugin.java [hunk 8]
```diff
@@ -0,0 +1,94 @@
...
+import net.runelite.client.plugins.PluginDescriptor;
+import net.runelite.api.NPC;
+import net.runelite.api.Client;
+import javax.inject.Inject;
+
+@PluginDescriptor
+(
+	name = "Kitten Notifier",
+	description = "Sends a notification when your kitten needs food, attention, or is grown.",
+	tags = {"kitten, notifications"}
```
Comment (Spedwards) [inline runelite-client/src/main/java/net/runelite/client/plugins/kittennotifier/KittenNotifierPlugin.java]:
```suggestion
	tags = {"kitten", "cat", "notifications"}
```

**Comment by ShaunDreclin** [comment]
Not sure if this is really necessary since we have the chat notifier plugin already, although I guess your kitten growing into a cat doesn't send a chat message.

**Pull request closed without merging**